crates/database/src/repositories/response.rs

```diff
@@ -18,6 +18,102 @@ impl PgResponseRepository {
     pub fn new(pool: DbPool) -> Self {
         Self { pool }
     }
+
+    /// Internal helper: find or create the hidden structural root response
+    /// ("root_response") for a given conversation.
+    ///
+    /// - Root responses are marked with metadata: { "root_response": true }
+    ///   and model "root_response".
+    /// - They never have response_items attached; they're purely structural.
+    async fn get_or_create_root(
+        &self,
+        conversation_uuid: Uuid,
+        workspace_id: &WorkspaceId,
+        api_key_id: &uuid::Uuid,
+    ) -> Result<Uuid, RepositoryError> {
+        // First, try to find an existing root for this conversation
+        let existing_row = retry_db!("get_root_response", {
+            let client = self
+                .pool
+                .get()
+                .await
+                .context("Failed to get database connection")
+                .map_err(RepositoryError::PoolError)?;
+
+            client
+                .query_opt(
+                    r#"
+                    SELECT id
+                    FROM responses
+                    WHERE conversation_id = $1
+                      AND workspace_id = $2
+                      AND COALESCE((metadata->>'root_response')::boolean, false) = true
```

Comment:
Query uses COALESCE((metadata->>'root_response')::boolean, false) = true but index uses metadata->>'root_response' = 'true'. PostgreSQL needs exact match to use partial index. Update query to:
`metadata->>'root_response' = 'true'`